apps/omg/test/omg/crypto_test.exs

```diff
@@ -24,58 +25,72 @@ defmodule OMG.CryptoTest do
   alias OMG.State.Transaction
   alias OMG.TypedDataHash
 
-  test "sha3 library usage, address generation" do
-    # test vectors below were generated using pyethereum's sha3 and privtoaddr
-    priv = Crypto.hash(<<"11">>)
-    py_priv = "7880aec93413f117ef14bd4e6d130875ab2c7d7d55a064fac3c2f7bd51516380"
-    py_pub = "c4d178249d840f548b09ad8269e8a3165ce2c170"
-    assert {:ok, ^priv} = Base.decode16(py_priv, case: :lower)
-    {:ok, pub} = DevCrypto.generate_public_key(priv)
-    {:ok, address} = Crypto.generate_address(pub)
-    assert {:ok, ^address} = Base.decode16(py_pub, case: :lower)
-  end
+  describe "recover_address/2" do
+    # Tests that we can digest, sign, and recover.
+    test "recovers address of the signer from a binary-encoded signature" do
+      {:ok, priv} = DevCrypto.generate_private_key()
+      {:ok, pub} = DevCrypto.generate_public_key(priv)
+      {:ok, address} = Crypto.generate_address(pub)
 
-  test "signature compatibility" do
-    msg = Crypto.hash("1234")
-    priv = Crypto.hash("11")
-    {:ok, pub} = DevCrypto.generate_public_key(priv)
-    {:ok, _} = Crypto.generate_address(pub)
-    # this test vector was generated using plasma.utils.utils.sign/2 from plasma-mvp
-    py_signature =
-      "b8670d619701733e1b4d10149bc90eb4eb276760d2f77a08a5428d4cbf2eadbd656f374c187b1ac80ce31d8c62076af26150e52ef1f33bfc07c6d244da7ca38c1c"
-
-    sig = DevCrypto.signature_digest(msg, priv)
-    assert ^sig = Base.decode16!(py_signature, case: :lower)
-  end
+      msg = :crypto.strong_rand_bytes(32)
+      sig = DevCrypto.signature_digest(msg, priv)
+
+      assert {:ok, ^address} = Crypto.recover_address(msg, sig)
+    end
+
+    # Test that we can sign and verify
+    test "recovers address from an encoded transaction" do
+      {:ok, priv} = DevCrypto.generate_private_key()
+      {:ok, pub} = DevCrypto.generate_public_key(priv)
+      {:ok, address} = Crypto.generate_address(pub)
+
+      raw_tx = Transaction.Payment.new([{1000, 1, 0}], [])
+      signature = DevCrypto.signature(raw_tx, priv)
+      assert byte_size(signature) == 65
+
+      assert true ==
+               raw_tx
+               |> TypedDataHash.hash_struct()
+               |> Crypto.recover_address(signature)
+               |> (&match?({:ok, ^address}, &1)).()
 
-  test "digest sign, recover" do
-    {:ok, priv} = DevCrypto.generate_private_key()
-    {:ok, pub} = DevCrypto.generate_public_key(priv)
-    {:ok, address} = Crypto.generate_address(pub)
-    msg = :crypto.strong_rand_bytes(32)
-    sig = DevCrypto.signature_digest(msg, priv)
-    assert {:ok, ^address} = Crypto.recover_address(msg, sig)
+      assert false ==
```

Comment:
You can use `refute` here (and get rid of the `false`), but that's up for discussion, depending on which style we want to go with.